syntax = "proto3";
package object_micro_app;





/*
SearchMicroAppList请求
*/
message SearchMicroAppListRequest {
message Query {
    /*
    按objectId列表查询
    */
    repeated string objectId = 1 ;
}
        
    
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 page_size = 2 ;
    /*
    查询条件
    */
    Query query = 3 ;
}

/*
SearchMicroAppList返回
*/
message SearchMicroAppListResponse {
message List {
    /*
    关联关系唯一名称
    */
    string name = 1 ;
    /*
    模型id
    */
    string objectId = 2 ;
    /*
    小产品Id
    */
    string microAppId = 3 ;
    /*
    小产品名称
    */
    string microAppName = 4 ;
    /*
    小产品首页
    */
    string homepage = 5 ;
    /*
    子页面相对路径
    */
    string subPath = 6 ;
    /*
    排序序号
    */
    int32 order = 7 ;
    /*
    状态
    */
    string status = 8 ;
    /*
    是否是默认小产品
    */
    string default = 9 ;
    /*
    标签
    */
    repeated string tags = 10 ;
}
        
    
    /*
    返回总数
    */
    int32 total = 1 ;
    /*
    页数
    */
    int32 page = 2 ;
    /*
    该页大小
    */
    int32 page_size = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
SearchMicroAppListApi返回
*/
message SearchMicroAppListResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    object_micro_app.SearchMicroAppListResponse data = 4 ;
}

